src/Core/src/Handlers/Image/ImageHandler.Android.cs

```diff
@@ -48,12 +48,16 @@ public static void MapBackground(IImageHandler handler, IImage image)
 		public static void MapSource(IImageHandler handler, IImage image) =>
 			MapSourceAsync(handler, image).FireAndForget(handler);
 
-		public static Task MapSourceAsync(IImageHandler handler, IImage image)
+		public static async Task MapSourceAsync(IImageHandler handler, IImage image)
 		{
-			return handler
+			await handler
 				.SourceLoader
-				.UpdateImageSourceAsync()
-				.ContinueWith((action) => handler.UpdateValue(nameof(IImage.IsAnimationPlaying)));
+				.UpdateImageSourceAsync();
+
+			// Because this resolves from a task we should validate that the
+			// handler hasn't been disconnected
+			if (handler.IsConnected())
+				handler.UpdateValue(nameof(IImage.IsAnimationPlaying));
```

Comment:
This was occasionally causing a crash in our device tests.

Also, I don't think we ever should have used ContinueWith here, I don't think `ContinueWith` will resolve on the synchronization context. I'm pretty sure the code inside `ContinueWith` will be on a task pool thread unless you pass `TaskScheduler.FromCurrentSynchronizationContext()`